webrtcclient.js

```diff
@@ -196,4 +196,33 @@ WebRTCClient.prototype.getStats = function() {
   });
 };
 
+WebRTCClient.prototype.waitForAudio(silence) {
```

Comment:
Taking a bool here gives you a basic test, but not much flexibility. May I suggest taking a callback instead? That's what we started doing in Firefox tests and it has proven very useful.

You pass the callback the array with frequency data, and it can figure out the index for a given frequency if you want to test across the whole spectrum.

It also allows you to do multiple tests at the callsite with the same analyser (either set up the analyser separately from the `waitForAudio` function, or let the `waitForAudio` callback drive the test steps). The former is what we use in Firefox tests, and allows you to return a promise and chain multiple tests for the same analyser.

See http://searchfox.org/mozilla-central/rev/d20e4431d0cf40311afa797868bc5c58c54790a2/dom/media/tests/mochitest/test_getUserMedia_mediaStreamTrackClone.html#107